src/StoreApi/Routes/Checkout.php

```diff
@@ -164,6 +164,32 @@ protected function get_route_post_response( \WP_REST_Request $request ) {
 		// Check order is still valid.
 		$order_controller->validate_order_before_payment( $order_object );
 
+		// Create a new user account as necessary.
+		if ( ! is_user_logged_in() && ( WC()->checkout()->is_registration_required() || ! empty( $request['create_account'] ) ) ) {
```

Comment:
I'm wondering if we should avoid using properties from the core checkout class due to it being so deeply coupled to the shortcode and checkout page. We could look this up from woo core settings so this API is not dependent on core checkout.

Reply:
Sounds good - will make those changes :)